pkg/agent/client.go

```diff
@@ -79,6 +80,58 @@ func newConnectionManager() *connectionManager {
 	}
 }
 
+// AgentAddress stores addresses that are reachable though an agent
+type AgentAddress struct {
+	IPv4        []string
+	IPv6        []string
+	Host        []string
+	CIDR        []string
+	FailureZone []string
+}
+
+type AddressType string
+
+const (
+	IPv4        AddressType = "ipv4"
+	IPv6        AddressType = "ipv6"
+	Host        AddressType = "host"
+	CIDR        AddressType = "cidr"
+	FailureZone AddressType = "failure-zone"
+)
+
+// GenAgentAddress generates an AgentAddress based on the input string, the
+// input string should be a comma-seprated list with each item in the format
+// of <addressType>=<address>
+func GenAgentAddress(addrs string) AgentAddress {
+	var agentAddr AgentAddress
+	entries := strings.Split(addrs, ",")
+	for _, entry := range entries {
+		kv := strings.Split(entry, "=")
+		if len(kv) != 2 {
+			continue
```

Comment:
We should at least print a warning/error if the configuration is incorrect

Reply:
warning message added